packages/plugin-aws-sdk/src/aws-sdk.ts

```diff
@@ -96,25 +105,33 @@ class AwsPlugin extends BasePlugin<typeof AWS> {
         }
     }
 
-    private _registerCompletedEvent(span: Span, request: AWS.Request<any, any>) {
+    private _registerCompletedEvent(span: Span, request: AWS.Request<any, any>, completedEventContext: Context) {
         const thisPlugin = this;
         request.on('complete', (response) => {
-            if (!request[thisPlugin.REQUEST_SPAN_KEY]) {
-                return;
-            }
-            request[thisPlugin.REQUEST_SPAN_KEY] = undefined;
-
-            if (response.error) {
-                span.setAttribute(AttributeNames.AWS_ERROR, response.error);
-            }
-
-            this._callUserResponseHook(span, response);
-            this.servicesExtensions.responseHook(response, span);
-
-            span.setAttributes({
-                [AttributeNames.AWS_REQUEST_ID]: response.requestId,
+            // The better alternative is to path the entire events of the event emitter,
```

Comment:
Since this is open source, could be nice to create a related issue and link to there, instead of this big comment

Reply:
Done. anything else or can i merge?